### DISCUSSION TITLE
Conundrum on forks

### DISCUSSION THREAD

**Discussion by sindizzy** [DotSpatial/DotSpatial#1458] [Q&A]
So I think I have some time to start contributing to V4 that's based on .NET Core. However, all my work-related projects are still based on .NET Framework and they will stay that way for at least a year or two. So I have a vested interest in doing some updates to the 2.x or 3.x line for my own use (and those others who wish to use my fork).

My fork at https://github.com/sindizzy/DotSpatial was based on the 2.0 release. I'm a bit confused as to how I can keep that fork and update it to the 3.x line and then make a different fork altogether for the 4.x line. When I try to create a brand new fork based on the current DS repo I get this message "No more forks can be created. These forks already exist:"

Any suggestions welcome. Thanks.

**Comment by jany-tenaj** [comment]
If I understood correctly you can fork DS to your own account only once. But there you can create forks from your main. Afterwards you'll have to pull the commit you want to work on. Then you can work on it and create a pull request that uses DS as upstream you want to push to.

Basically I'd say I'm glad I don't have to do that ;)

**Comment by sindizzy** [comment]
ok let me try that. thanks.

**Comment by sindizzy** [comment]
OK after some consultation here is what has been recommended: 

1. Fork the v4 repo in Github since I will likely want to make PRs and v4 is the only currently version that is accepting PRs.

2. Clone v3 repo (locally) under a different name, e.g. DotSpatialv3 and then create/push that to Github as a separate repo. This will mean that it will exist separately and not linked to the main DS repo.

So # 1 I think is no problem. I can do that right now. But # 2, I wanted to ask if that's allowed. I mean it's not my codebase but I need to keep it alive for my current projects (and those that wish to use it). I anticipate that within a year or two, I will switch all my projects to Core so its not a long term thing.

**Reply by jany-tenaj** [reply]
It's an MIT license. As far as I know you should be able to do this. Just go and read it ;)
